app/src/main/kotlin/com/k0d4black/theforce/features/character_details/CharacterDetailViewModel.kt

```diff
@@ -18,18 +16,23 @@ import kotlinx.coroutines.flow.collect
 import kotlinx.coroutines.launch
 
 internal class CharacterDetailViewModel(
+    application: Application,
     private val getSpeciesUseCase: SpeciesUseCase,
     private val getPlanetUseCase: PlanetUseCase,
     private val getFilmsUseCase: FilmsUseCase
-) : ViewModel() {
+) : AndroidViewModel(application) {
 
     val detailViewState: LiveData<CharacterDetailsViewState>
         get() = _detailViewState
 
     private var _detailViewState = MutableLiveData<CharacterDetailsViewState>()
 
-    private val characterDetailExceptionHandler = CoroutineExceptionHandler { _, exception ->
-        val message = ExceptionHandler.parse(exception)
+    val app by lazy {
+        application
+    }
+
+    private val characterDetailExceptionHandler = CoroutineExceptionHandler { context, exception ->
```

Comment:
Also, you can leave the underscore for unused parameters like context to have a consistent style for the lambdas, and lint will be silent about unused parameters.👍

```kotlin
private val characterDetailExceptionHandler = CoroutineExceptionHandler { _, exception->
}
```